docs/benchmarkdotnet.md

```diff
@@ -0,0 +1,332 @@
+* `-` in the Gen column means that no garbage collection was performed.
+
+## Profiling
+
+If you want to profile the benchmarked code, you can use some of the built-in profilers offered by BenchmarkDotNet:
+
+- [ETW Profiler](https://adamsitnik.com/ETW-Profiler/) - it profiles the benchmarked .NET code on Windows and exports the data to a trace file which can be opened with PerfView or Windows Performance Analyzer.
+- [Concurrency Visualizer Profile](https://adamsitnik.com/ConcurrencyVisualizer-Profiler/) - it profiles the benchmarked .NET code on Windows and exports the data to a trace file which can be opened with Concurrency Visualizer. The exported Trace file can be also opened with PerfView or Windows Performance Analyzer
+
+Commands that enable the profilers are: `--profiler ETW` and `--profiler CV`.
+
+After running the benchmarks, BenchmarkDotNet is going to print the path to the trace files:
+
+```log
+// * Diagnostic Output - EtwProfiler *
+Exported 1 trace file(s). Example:
+C:\Projects\performance\artifacts\20190215-0303-51368\Benchstone\BenchF\Adams\Test.etl
+```
+
+## Disassembly
+
+If you want to disassemble the benchmarked code, you need to use the [Disassembly Diagnoser](http://adamsitnik.com/Disassembly-Diagnoser/). It allows disassembling `asm/C#/IL` in recursive way on Windows for .NET and .NET Core (all Jits) and `asm` for Mono on any OS.
+
+You can do that by passing `--disassm` to the app or by using `[DisassemblyDiagnoser(printAsm: true, printSource: true)]` attribute or by adding it to your config with `config.With(DisassemblyDiagnoser.Create(new DisassemblyDiagnoserConfig(printAsm: true, recursiveDepth: 1))`.
+
+Example: `dotnet run -f netcoreapp2.0 -- --filter System.Memory.Span<Int32>.Reverse -d`
+
+```assembly
+; System.Runtime.InteropServices.MemoryMarshal.GetReference[[System.Byte, System.Private.CoreLib]](System.Span`1<Byte>)
+       sub     rsp,28h
+       cmp     qword ptr [rcx],0
+       jne     M00_L00
+       mov     rcx,qword ptr [rcx+8]
+       call    System.Runtime.CompilerServices.Unsafe.AsRef[[System.Byte, System.Private.CoreLib]](Void*)
+       nop
+       add     rsp,28h
+       ret
+M00_L00:
+       mov     rax,qword ptr [rcx]
+       cmp     dword ptr [rax],eax
+       add     rax,8
+       mov     rdx,qword ptr [rcx+8]
+       add     rax,rdx
+       add     rsp,28h
+       ret
+```
+
+## Multiple Runtimes
+
+The `--runtimes` or just `-r` allows you to run the benchmarks for **multiple Runtimes**.
+
+Available options are: Mono, CoreRT, net461, net462, net47, net471, net472, netcoreapp2.0, netcoreapp2.1, netcoreapp2.2, netcoreapp3.0.
```

Comment:
We should update the `TargetFrameworks` on the props file. Currently there is only net461 for Desktop.

Reply:
When it comes to `--runtimes` other options than `net461` are possible because here we specify the TFM of the auto-generated project. For example, we can generate a net472 project which references MicroBenchmarks.csproj which targets net461